### ISSUE TITLE
Run minikube in Travis CI

### ISSUE THREAD

**Issue by mlangbehn** [IBM/OpenWhisk-on-Kubernetes#21]
We have a stub created for testing all Pull Requests in `minikube`. With the change in apache/incubator-openwhisk-deploy-kube#5, we are able to run OpenWhisk in minikube. We need to implement this in our Traivs CI testing.

**Comment by mlangbehn** [comment]
It looks like this may not be possible. See failures in [this](https://travis-ci.org/mlangbehn/OpenWhisk-on-Kubernetes/builds/239749388#L389) Travis run:

```text
E0605 21:52:41.509372    4641 start.go:127] Error starting host: Error creating host: Error creating machine: Error in driver during machine creation: virError(Code=1, Domain=20, Message='internal error: no supported architecture for os type 'hvm'').
 Retrying.
E0605 21:52:41.509779    4641 start.go:133] Error starting host:  Error creating host: Error creating machine: Error in driver during machine creation: virError(Code=1, Domain=20, Message='internal error: no supported architecture for os type 'hvm'')
```

**Comment by Tomcli** [comment]
Oh, is your OpenWhisk working on Minikube? Right now after I changed the API version to 1.23 and completed the openwhisk config job, my Invoker failed to ping the controller when I try to make a wsk action.

**Comment by mlangbehn** [comment]
@Tomcli: I wasn't able to get the config job to complete in my testing, so I figured I would start working on getting Travis testing to work in my [add_minikube_tests](https://github.com/mlangbehn/OpenWhisk-on-Kubernetes/tree/add_minikube_tests) branch. Based on my testing (see log output in my [last comment](https://github.com/IBM/OpenWhisk-on-Kubernetes/issues/21#issuecomment-306335974)), Minikube is not going to work in Travis CI with KVM as the `docker-machine-driver`. I can say from past experience that VirtualBox doesn't work in Travis CI.

Today, I plan on looking into alternative options.

**Comment by bbrowning** [comment]
@Tomcli You need to `minikube ssh` and `sudo ip link set docker0 promisc on` I believe. Several members of my team have successfully run OpenWhisk on both minikube and minishift if you need additional help debugging things.

**Comment by mlangbehn** [comment]
@bbrowning: The output from Travis CI shows: `internal error: no supported architecture for os type 'hvm'` when running `minikube start --vm-driver kvm`, and I know from past attempts that VirtualBox is not an option in Travis CI.

In the [apache/incubator-openwhisk-deploy-kube](https://github.com/apache/incubator-openwhisk-deploy-kube), I found that this is a problem that has already been solved. I am about to start testing with a configuration based of the scripts in [apache/incubator-openwhisk-deploy-kube's .travis.yml](https://github.com/apache/incubator-openwhisk-deploy-kube/blob/master/.travis.yml) and the scripts in [apache/incubator-openwhisk-deploy-kube's .travis/](https://github.com/apache/incubator-openwhisk-deploy-kube/tree/master/.travis).

**Comment by Tomcli** [comment]
@bbrowning Thank you, that fixes my problem. I think we can start adding instructions for Minikube then.

**Comment by bbrowning** [comment]
@mlangbehn Yes - I'm not aware of any way to get virtual machines running on top of Travis CI. That may be a blocker for running minikube itself in a Travis job.

**Comment by mlangbehn** [comment]
Branch where I am working to get Travis CI testing on all branches and pull requests:
https://github.com/mlangbehn/OpenWhisk-on-Kubernetes/tree/split_out_travis_install_script

**Comment by ivan4th** [comment]
May I suggest that you try [kubeadm-dind-cluster](https://github.com/Mirantis/kubeadm-dind-cluster)? It works on Travis and CircleCI rather well (it has it's own test suite running on Travis plus we're using it to run [Virtlet](https://github.com/Mirantis/virtlet) tests). Unlike Minikube it doesn't need VMs and unlike `hack/local-up-cluster.sh` you don't have to build Kubernetes from source.

**Comment by diemol** [comment]
Hi, I found this issue by coincidence (I am now looking how to run MiniShift in Travis).

Anyway, I found this [script](https://github.com/zalando/zalenium/blob/master/kubernetes/minikube-ci-initialize.sh), that I adapted and it lets you run Minikube in Travis.
Give it a try :)
